network/network_interface.h

```diff
@@ -0,0 +1,18 @@
+#include "util.h"
+
+typedef struct {
+  void (*power)(bool on);
+  void (*link_change)(uint8_t port_index, bool is_up);
+  size_t (*receive)(uint8_t port_index, uint8_t *data, size_t length);
+} NetworkInterfaceCallbacks;
```

Comment:
These are the 3 callbacks from the driver back into the application.

- `power` needs to power on or off the adin chip if the board has that capability — we will set ADIN_RST high/low in this function
- `link_change` is the notification that one of the ports went up or down
- `receive` is where we receive data from the network